spiped/main.c

```diff
@@ -241,10 +241,15 @@ main(int argc, char * argv[])
 
 	/* Figure out where our pid should be written. */
 	if (opt_p == NULL) {
```

Comment:
I'm tempted to write this now as

```
if (asprintf(&pidfilename, opt_p ? "%s" : "%s.pid", opt_p ? opt_p : opt_s)) {
    warnp("asprintf");
    goto err0;
}
```

but maybe you'll think that's too obscure?

Reply:
I had something similar to that in a draft, but I did think it was unnecessarily un-obvious.  But if you prefer it that way, I won't object.  I'll upload a new draft.